haystack/core/pipeline/pipeline.py

```diff
@@ -378,6 +391,8 @@ def run(  # noqa: PLR0915, PLR0912, C901, pylint: disable=too-many-branches
 
                 if component_pipeline_outputs:
                     pipeline_outputs[component_name] = deepcopy(component_pipeline_outputs)
+                    if component_name in include_outputs_from:
+                        intermediate_outputs[component_name] = deepcopy(component_pipeline_outputs)
```

Comment:
Just to clarify here was the original request to only include stuff from `include_outputs_from` or also include all outputs from the pipeline? 

I would have thought it would make the most sense to include all values in `pipeline_outputs`. At least that's my view on what intermediate outputs are since those are all outputs I'd expect to see up until the break point. 

I realize we return `pipeline_outputs` in the Breakpoint Exception but I think the utility would be also to have them in the snapshot. This way we can also reset/resume the `pipeline_outputs` to the correct state when we restart the Pipeline.

Reply:
> Just to clarify here was the original request to only include stuff from include_outputs_from or also include all outputs from the pipeline?

The requested refers to only those components specified in the `include_outputs_from`

> I realize we return pipeline_outputs in the Breakpoint Exception but I think the utility would be also to have them in the snapshot. This way we can also reset/resume the pipeline_outputs to the correct state when we restart the Pipeline.

hmmm...I also noticed now that we only return it in the Exception and not on the snapshot - we can included it in the snapshot and probably then remove the `include_outputs_from`